app/models/manageiq/providers/openstack/cloud_manager/vm/resize.rb

```diff
@@ -1,9 +1,18 @@
 module ManageIQ::Providers::Openstack::CloudManager::Vm::Resize
-  def validate_resize
-    msg = validate_vm_control
-    return {:available => msg[0], :message => msg[1]} unless msg.nil?
-    return {:available => true, :message => nil} if %w(ACTIVE SHUTOFF).include?(raw_power_state)
-    {:available => false, :message => _("The Instance cannot be resized, current state has to be active or shutoff.")}
+  extend ActiveSupport::Concern
+
+  included do
+    supports :resize do
+      msg = validate_vm_control
```

Comment:
And just for completeness

```ruby
  def validate_shutdown_guest
    msg = validate_vm_control
    return {:available => msg[0], :message => msg[1]} unless msg.nil?
    return {:available => true,   :message => ''}     if tools_status && tools_status == 'toolsNotInstalled'
    return {:available => true,   :message => nil}    if current_state == 'on'
    {:available => false,  :message => 'The VM is not powered on'}
  end
```

```ruby
supports :shutdown do
  unsupported_reason_add(:shutdown, unsupported_reason(:control)) unless supports_control?
  unsupported_reason_add(:shutdown, _("Tools not installed")) if tools_status == 'toolsNotInstalled'
  unsupported_reason_add(:shutdown, _("The VM is not powered on")) if current_state == 'on'
end
```

please let me know if there are still unclear things